### PR TITLE
Remove old-style xva import code (finally)

### PR DESCRIPTION
**Pull request by robhoes**
Cleaning up some very old, very dead code ("old-style" means "from before 2007").

### CODE REVIEW

**Comment by robhoes** [comment]
Still running some tests, just to be sure.

File: ocaml/xapi-cli-server/cli_operations.ml [hunk 1]
```diff
@@ -5363,291 +5363,52 @@ let vm_import fd printer rpc session_id params =
...
+              Constants.import_metadata_uri
           else
-            uuids
-        in
-        marshal fd (Command (Print (String.concat "," uuids)))
-    | _ ->
-        failwith "Thin CLI protocol error"
+            Constants.import_uri
+          )
+          (Ref.string_of session_id) (Ref.string_of task_id) full_restore
```
Comment (lindig) [inline ocaml/xapi-cli-server/cli_operations.ml:5377]: This code assumes that no URL escaping is ever required. Is this true? I'm no too fond of constructing a URI this way and would have preferred constructing a map or assoc list which then some function turns into a URI, taking care of escaping.

File: ocaml/xapi/import.ml [hunk 2]
```diff
@@ -2169,121 +2169,100 @@ let metadata_handler (req : Request.t) s _ =
...
+              header.version config.force vdis
+          in
+          (* CA-48768: Stream_vdi.recv_all only checks for task cancellation
+             						   every ten seconds, so we need to check again now. After this
+             						   point, we disable cancellation for this task. *)
+          TaskHelper.exn_if_cancelling ~__context ;
+          TaskHelper.set_not_cancellable ~__context ;
+          (* Pre-miami GA exports have a checksum table at the end of the export. Check the calculated checksums *)
+          (* against the table here. Nb. Rio GA-Miami B2 exports get their checksums checked twice! *)
+          ( if header.version.export_vsn < 2 then
```
Comment (lindig) [inline ocaml/xapi/import.ml:2230]: This is some deeply nested code. Could this be made flatter?

File: ocaml/xapi-cli-server/cli_operations.ml (hunk 1 above)
Comment (robhoes) [reply ocaml/xapi-cli-server/cli_operations.ml:5377]: This is not new code. This PR mostly removes and reformats code.

Comment (robhoes) [reply ocaml/xapi-cli-server/cli_operations.ml:5377]: This view makes it easier to see: https://github.com/xapi-project/xen-api/pull/4539/files?diff=split&w=1

File: ocaml/xapi/import.ml (hunk 2 above)
Comment (robhoes) [reply ocaml/xapi/import.ml:2227]: I guess the next step might be to zap the code that handles pre-miami exports 🤣

**Comment by robhoes** [comment]
@djs55 @jonludlam Can you believe that we are doing this?!

**Comment by johnelse** [comment]
I always wondered if anyone used this stuff :D

**Comment by jonludlam** [comment]
mumble mumble transfer vm...?

**Comment by jonludlam** [comment]
or perhaps the venerable debian vm?

**Comment by robhoes** [comment]
Jon is right! It was the Transfer VM!

**Review by djs55 — APPROVED** [review]
LGTM ;)

**Comment by djs55** [comment]
Did this include the logic to include empty blocks in the output stream in case a long sequence of zeros causes the TCP stream to go idle long enough to be silently terminated by middleboxes? I enjoyed that one.

File: ocaml/xapi/import.ml (hunk 2 above)
Comment (avsm) [reply ocaml/xapi/import.ml:2227]: I think I might have some Rio exports lying around... perhaps I should quietly destroy them ;-)

**Review by psafont — APPROVED** [review]
The removal of code looks reasonable, especially how the CLI protocol is handled.

I'm also happy an unverified TLS connection got zapped

**Comment by robhoes** [comment]
@djs55 I think that you mean this one: https://github.com/xapi-project/xen-api/commit/bd54ec89438ed1358ac8ba02bf60110c0f71a313. This code is still there! :)

**Review by edwintorok — APPROVED** [review]

**Comment by edwintorok** [comment]
I assume you've tested and regular xva import still works with the new `xe`.

**Pull request merged**